### PR TITLE
ROC-4696 Replace anchor links with macros

### PR DESCRIPTION
**Pull request by andrewwa-kainos**
### JIRA link
https://tools.hmcts.net/jira/browse/ROC-4696

### Change description
Replace (most) anchor links with internal and external link macros

### Work checklist

- [ ] Unit tests added where applicable
- [ ] Route tests added for new pages
- [ ] New pages included in a11y tests
- [ ] Task list and task completeness checks updated
- [ ] Check and send page updated
- [ ] Routes, page content and page flows of new features are toggled 
- [ ] UI changes look good on mobile
- [ ] Required Google Analytics events are being sent 

### Developer self-QA run statement

- [x] I have clicked through the running application to see if all changes I made actually work.

### Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: src/main/features/claimant-response/views/confirmation.njk [hunk 1]
```diff
@@ -72,9 +73,8 @@
         <p>{{ t('A mediator will contact you and the defendant in the next 14 days to arrange an appointment.') }}</p>
       {% elseif claim.claimantResponse.type === ClaimantResponseType.REJECTION and claim.directionsQuestionnaireDeadline %}
         <p>
-          {{ t('You need to')}}
-          <a href="{{ DashboardPaths.directionsQuestionnairePage.evaluateUri({ externalId: claim.externalId }) }}">
-            {{ t('complete a directions questionnaire form ')}}</a> {{ t('to tell us more about the claim.')}}
+          {{ t('You need to') }}
+          {{ internalLink(t('complete a directions questionnaire form'), DashboardPaths.directionsQuestionnairePage.evaluateUri({ externalId: claim.externalId })) }}.
```
Comment (andrewwa-kainos) [inline src/main/features/claimant-response/views/confirmation.njk]: Missing the end of the sentence

File: src/main/features/first-contact/views/access-denied.njk [hunk 2]
```diff
@@ -1,12 +1,13 @@
...
+{% from "internalLink.njk" import internalLink %}
 
 {% set heading = 'You are not authorised to view the claim!' %}
 
 {% block content %}
   <div class="grid-row">
     <div class="column-two-thirds">
       <p>{{ t('There was an error during your last operation. Please try again later.') }}</p>
-      <a href="/first-contact/claim-reference">{{ t('Click here to try again') }}</a>
+      {{ internalLink(t('Click here to try again'), '/first-contact/claim-reference') }}
```
Comment (robertparkinson) [inline src/main/features/first-contact/views/access-denied.njk]: can you please move this path into a Path file?

File: src/main/features/first-contact/views/start.njk [hunk 3]
```diff
@@ -36,11 +38,9 @@
         <h2 class="heading-medium" id="subsection-title">{{ t('Support') }}</h2>
         <nav role="navigation" aria-labelledby="subsection-title">
           <ul class="font-xsmall">
-            <li><a href="https://www.gov.uk/make-court-claim-for-money/court-fees">{{ t('Money claim fees') }}</a></li>
-            <li><a href="/other-options">{{ t('Other options') }}</a></li>
-            <li><a class="external" rel="external"
-                   href="https://courttribunalfinder.service.gov.uk/courts/county-court-business-centre-ccbc"
-                   target="_blank">{{ t('Contact us') }}</a></li>
+            <li>{{ externalLink(t('Money claim fees'), 'https://www.gov.uk/make-court-claim-for-money/court-fees') }}</li>
+            <li>{{ internalLink(t('Other options'), '/other-options') }}</li>
```
Comment (robertparkinson) [inline src/main/features/first-contact/views/start.njk]: this should also be in a Path file. I'm wondering if we should put the external ones in one common Paths file as well? might be best to keep that for another story though

File: src/main/views/forbidden.njk [hunk 4]
```diff
@@ -1,11 +1,14 @@
...
 
 {% block content %}
   <p>
-    {{ t('You can return to <a href="/">Civil Money Claims</a> and start your claim or defence again.') | safe }}
+    {{ t('You can return to {{ homepage }} and start your claim or defence again.',
+      { homepage: internalLink(t('Civil Money Claims'), '/') }) | safe }}
   </p>
 
-  <a class="button" href="/">{{ t('Start again') }}</a>
+  {{ internalLink(t('Start again'), '/', 'button') }}
```
Comment (robertparkinson) [inline src/main/views/forbidden.njk]: Another one to move into a Path file.

File: src/main/views/includes/propositional_navigation.njk [hunk 5]
```diff
@@ -1,12 +1,16 @@
...
     <nav id="proposition-menu">
-      <a href="/" id="proposition-name">{% if serviceName %}{{ t(serviceName) }}{% endif %}</a>
+      {% if serviceName %}
+        {{ internalLink(t(serviceName), '/', id='proposition-name') }}
+      {% endif %}
       <ul id="proposition-links" class="pull-right">
         {% if isLoggedIn %}
-          <li><a href="/dashboard" >{{ t('My account') }}</a></li>
-          <li><a href="/logout">{{ t('Sign out') }}</a></li>
+          <li>{{ internalLink(t('My account'), '/dashboard') }}</li>
```
Comment (robertparkinson) [inline src/main/views/includes/propositional_navigation.njk]: Move to Path file as above for both of these please.

File: src/main/views/includes/support-links.njk [hunk 6]
```diff
@@ -1,22 +1,24 @@
...
+      {{ internalLink(t('Help using GOV.UK'), 'https://www.gov.uk/help') }}
     </li>
     <li>
-      <a href="{{ AppPaths.contactUsPage.uri }}">{{ t('Contact us') }}</a>
+      {{ internalLink(t('Contact us'), AppPaths.contactUsPage.uri) }}
     </li>
   {% if featureToggles.testingSupport %}
     <li>
-      <a href="/testing-support">Testing support</a>
+      {{ internalLink(t('Testing support'), '/testing-support') }}
```
Comment (robertparkinson) [inline src/main/views/includes/support-links.njk]: Move to Path file please

File: src/main/views/not-found.njk [hunk 7]
```diff
@@ -1,11 +1,14 @@
...
 
 {% block content %}
   <p>
-    {{ t('You can return to <a href="/">Civil Money Claims</a> and start your claim or defence again.') | safe }}
+    {{ t('You can return to {{ homepage }} and start your claim or defence again.',
+      { homepage: internalLink(t('Civil Money Claims'), '/') }) | safe }}
   </p>
 
-  <a class="button" href="/">{{ t('Start again') }}</a>
+  {{ internalLink(t('Start again'), '/', 'button') }}
```
Comment (robertparkinson) [inline src/main/views/not-found.njk]: Use entry from Path file created for above instance

File: src/main/features/first-contact/views/start.njk (hunk 3 above)
Comment (andrewwa-kainos) [reply src/main/features/first-contact/views/start.njk]: The `Other options` link goes to a missing page. Waiting for feedback from UX guys on it.

Comment (andrewwa-kainos) [reply src/main/features/first-contact/views/start.njk]: This version of the first-contact start page is only for local use - the live system will point users to a GOV.UK page instead. Removed the dead link.

**Pull request merged**